### PR TITLE
allow app-startup-mode to be specified on CLI or in env

### PR DESCRIPTION
**Pull request by maxtaco**
- also change semantics, so that in a config file, `app_status_mode : ""` now means "default"
  which is "service"

### CODE REVIEW

**Comment by maxtaco** [comment]
bump, can i get this in? it's helping me debug CORE-3285

File: go/libcmdline/cmdline.go [hunk 1]
```diff
@@ -343,6 +347,10 @@ func (p *CommandLine) PopulateApp(addHelp bool, extraFlags []cli.Flag) {
 			Usage: "Specify an alternate API URI path prefix.",
 		},
 		cli.StringFlag{
+			Name:  "app-start-mode",
+			Usage: "specify 'service' to auto-start UI app, or anything else to disable",
```
Comment (gabriel) [inline go/libcmdline/cmdline.go]: Maybe title case (like above).. and maybe fix to be consistent below.

I would say:
```
Specify 'service' to automatically start the UI app when the service starts up or 'disabled' to disable it
```

File: go/libkb/env.go [hunk 2]
```diff
@@ -1040,8 +1040,29 @@ func (e *Env) GetMountDir() (string, error) {
 	}
 }
 
+func ParseAppStartMode(s string) AppStartMode {
+	switch s {
+	case "":
+		return AppStartModeDefault
+	case "service":
+		return AppStartModeService
+	default:
```
Comment (gabriel) [inline go/libkb/env.go]: case "disabled":

(and then return error if parse fails)?

File: go/libkb/env.go [hunk 3]
```diff
@@ -1040,8 +1040,29 @@ func (e *Env) GetMountDir() (string, error) {
 	}
 }
 
+func ParseAppStartMode(s string) AppStartMode {
```
Comment (gabriel) [inline go/libkb/env.go]: Would be nice to return error if the string was invalid.

I guess in the past we'd have an enum for the error case.. e.g. AppStartModeError.

**Comment by gabriel** [comment]
Comments, otherwise 👍

**Comment by maxtaco** [comment]
thanks for the review!

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 392506b32f4a...ba096ceb6fed introduced some visual changes on linux:
 * Changed: **unlock_folders-Success** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1467381971252-392506b32f4a-ba096ceb6fed-linux/unlock_folders-Success-392506b32f4a.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1467381971252-392506b32f4a-ba096ceb6fed-linux/unlock_folders-Success-ba096ceb6fed.png)  
   unlock_folders-Success comparison

**Pull request merged**